subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -359,6 +359,7 @@ public TransformServiceLookup(Provider<FileSystemLocation> inputFileProvider, @N
                 DeprecationLogger
                     .deprecate("Injecting the input artifact of a transform as a File")
                     .withAdvice("Declare the input artifact as Provider<FileSystemLocation> instead.")
+                    .withDslReferenceForType(TransformAction.class)
```

Comment:
There might be a better fit for this one or we should add one. I think this is deprecating a behavior and not just an API

Reply:
Noted and leaving undocumented for now.